### PR TITLE
prototypes fallback

### PR DESCRIPTION
**Pull request by mejango**
If no tier is specified and the contract's `shouldMintByDefault` is true, the best available tier for the contribution price will be minted.

### CODE REVIEW

File: contracts/JBTieredLimitedNFTRewardDataSource.sol [hunk 1]
```diff
@@ -155,9 +161,9 @@ contract JBTieredLimitedNFTRewardDataSource is
 
     @return The tier number of the specified token ID.
   */
-  function tierIdOfToken(uint256 _tokenId) public view override returns (uint256) {
-    // The tier ID is in the first 32 bytes.
-    return uint256(uint32(_tokenId));
+  function tierIdOfToken(uint256 _tokenId) public pure override returns (uint256) {
+    // The tier ID is in the first 8 bytes.
```
Comment (drgorillamd) [inline contracts/JBTieredLimitedNFTRewardDataSource.sol:165]: bits

File: contracts/JBTieredLimitedNFTRewardDataSource.sol [hunk 2]
```diff
@@ -312,35 +321,98 @@ contract JBTieredLimitedNFTRewardDataSource is
...
+    for (uint256 _i = _numTiers; _i != 0; ) {
+      // Set the tier being iterated on. Tier's are 1 indexed.
+      _tier = tiers[_i];
+
+      // Mint if the contribution value is at least as much as the floor, there's sufficient supply.
+      if (_tier.contributionFloor <= _amount && _tier.remainingQuantity != 0) {
+        // Mint the tokens.
+        uint256 _tokenId = _mintForTier(_i, _tier, _beneficiary);
+
+        emit Mint(_tokenId, _i, _beneficiary, _amount, 0, msg.sender);
```
Comment (drgorillamd) [inline contracts/JBTieredLimitedNFTRewardDataSource.sol:370]: Quantity should still be 1 (instead of 0)?

Comment (mejango) [reply contracts/JBTieredLimitedNFTRewardDataSource.sol:370]: 0 gives us some insight that this was triggered without specifying a reward tier

File: contracts/JBTieredLimitedNFTRewardDataSource.sol [hunk 3]
```diff
@@ -68,10 +74,10 @@ contract JBTieredLimitedNFTRewardDataSource is
   */
   function allTiers() external view override returns (JBNFTRewardTier[] memory _tiers) {
     // Initialize an array with the appropriate length.
-    _tiers = new JBNFTRewardTier[](numberOfTiers);
+    _tiers = new JBNFTRewardTier[](numTiers);
 
     // Loop through each tier.
-    for (uint256 _i = 0; _i < numberOfTiers; ) {
+    for (uint256 _i = 0; _i < numTiers; ) {
```
[the remark this answers is not in the document]
Comment (mejango) [reply contracts/JBTieredLimitedNFTRewardDataSource.sol:80]: lol. there was numTiers written somewhere so i converged the two. down with either.

**Review by drgorillamd — APPROVED** [review]

**Pull request merged**